flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraphBuilder.java

```diff
@@ -221,7 +222,7 @@ public static ExecutionGraph buildGraph(
 				vertex.initializeOnMaster(classLoader);
 			}
 			catch (Throwable t) {
-					throw new JobExecutionException(jobId,
+					throw new JobExecutionException(jobId, ApplicationStatus.fromJobStatus(executionGraph.getState()),
```

Comment:
I think it is not correct to already decide on the `ApplicationStatus` here. If the initialization of the job fails, we might still be able to retry it. Maybe it would be better to not use the `JobExecutionException` across the client and server. Maybe it should stay a client side concept which is only thrown by the client when obtaining the job result.

Reply:
@tillrohrmann , what's the problem when using the `JobExecutionException` across the client and server?

Reply:
One problem is that we are specifying the "final" application status for the `JobExecutionException` here. I think this is wrong since `executionGraph.getState()` could also be something different than a globally terminal state.